src/main/java/main/model/db/dao/project/UserDao.java

```diff
@@ -39,6 +67,36 @@ public UserDto IsAuthorized(String sessionId) throws AqualityException {
         else{
             throw new AqualityPermissionsException("Credentials you've provided are not valid. Reenter please.", user);
         }
+
+        ProjectUserDto projectUserDto = new ProjectUserDto();
+        projectUserDto.setUser_id(user.getId());
+        user.setProjectUsers(new ProjectUserController(user).getProjectUserForPermissions(projectUserDto));
         return user;
     }
+
+    /**
+     * Check If Token Valid
+     * @param sessionId Session ID = project:{projectId}:{token}
+     * @return Project Engineer
+     * @throws AqualityException error about authorization status
+     */
+    private UserDto IsAuthorizedToken(String sessionId) throws AqualityException {
+        String[] strings = sessionId.split(":");
+        Integer projectId =  Integer.parseInt(strings[1]);
+        boolean isTokenValid = new ImportTokenController(new UserDto()).isTokenValid(strings[2],projectId);
+
+        if(isTokenValid){
+            UserDto user = new UserDto();
+            ProjectUserDto projectUser = new ProjectUserDto();
+            projectUser.setProject_id(projectId);
+            projectUser.setViewer(1);
+            projectUser.setEngineer(1);
+            user.setProjectUsers(Collections.singletonList(projectUser));
+            user.setApiSessionProjectId(projectId);
+            return user;
+        }
+        else{
+            throw new AqualityPermissionsException("The import token is not valid", null);
```

Comment:
maybe instead of null better to use new Exception()?

Reply:
null means this is not real user but an anonymous user